modules/extensions-support/src/main/java/org/mule/module/extension/internal/runtime/OperationContextAdapter.java

```diff
@@ -26,4 +27,14 @@
      * an operation is to be executed
      */
     MuleEvent getEvent();
+
+    /**
+     * Returns an object which is configuring the operation being executed. The actual type
+     * of the instance is unknown, but it's guaranteed to be a realisation of the {@link Configuration}
+     * model that was set for the operation
+     *
+     * @param <C> the generic type of the configuration instance
+     * @return an {@link Object} consistent with a corresponding {@link Configuration} model
+     */
+    <C> C getConfigurationInstance();
```

Comment:
Right now, your implementation is the only one and it already need this method (in OperationContextAdapter). 
For me, this method must go in this interface, not in the adapter.